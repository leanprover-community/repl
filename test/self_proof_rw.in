{"cmd": "theorem self_application : 1 = 0 := by rw [self_application]"}

{"cmd": "theorem self_application : 1 = 0 := by sorry", "sorries": "individual"}

{"proofState": 0, "tactic": "rw [self_application]"}
