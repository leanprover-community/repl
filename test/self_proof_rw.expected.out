{"messages":
 [{"severity": "error",
   "pos": {"line": 1, "column": 8},
   "endPos": {"line": 1, "column": 24},
   "data":
   "fail to show termination for\n  self_application\nwith errors\nfailed to infer structural recursion:\nno parameters suitable for structural recursion\n\nwell-founded recursion cannot be used, `self_application` does not take any (non-fixed) arguments"}],
 "env": 0}

{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 39},
   "goal": "‚ä¢ 1 = 0",
   "endPos": {"line": 1, "column": 44}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 8},
   "endPos": {"line": 1, "column": 24},
   "data": "declaration uses 'sorry'"}],
 "env": 1}

{"proofStatus":
 "Error: kernel type check failed: (kernel) declaration has free variables '[anonymous]'",
 "proofState": 1,
 "goals": []}

